MonstieWash/Assets/Scripts/Player/PlayerHand_AnimationController.cs

```diff
@@ -0,0 +1,86 @@
+using NUnit.Framework.Constraints;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class PlayerHand_AnimationController : MonoBehaviour
+{
+    private enum AnimState { Stationary, Idle, Grab, Pet };
+    private AnimState m_currentAnim = AnimState.Idle;
+
+    [Tooltip("PlayerHand stationary animation")][SerializeField] private AnimationClip anim_stationary;
+    [Tooltip("PlayerHand idle animation")][SerializeField] private AnimationClip anim_idle;
+    [Tooltip("PlayerHand grab animation")][SerializeField] private AnimationClip anim_grab;
+    [Tooltip("PlayerHand petting animation")][SerializeField] private AnimationClip anim_pet;
+    
+    private Dictionary<AnimState, AnimationClip> m_handAnims = new Dictionary<AnimState, AnimationClip>();
+
+    private float m_stillTime = 0f;   // Consecutive time spent still
+    private bool m_isMoving = false;    // Whether hand is moving this frame
+    private bool m_wasMoving = false;   // Whether had was moving last frame
+
+    private PlayerHand m_playerHand;
+    private Animation m_animator;
+    private ToolSwitcher m_toolSwitcher;
+
+    private void Awake()
+    {
+        m_playerHand = GetComponentInParent<PlayerHand>();
+        m_animator = GetComponent<Animation>();
+        m_toolSwitcher = GetComponentInParent<ToolSwitcher>();
+        SetupDictionary();
+    }
+
+    private void SetupDictionary()
+    {
+        m_handAnims.Add(AnimState.Stationary, anim_stationary);
+        m_handAnims.Add(AnimState.Idle, anim_idle);
+        m_handAnims.Add(AnimState.Grab, anim_grab);
+        m_handAnims.Add(AnimState.Pet, anim_pet);
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        TrackIdleTime();
+
+        // Arranged in order of priority (only 4-5 hand animations, no need to overengineer?)
+        if (!m_toolSwitcher.HandFree)
+        {
+            PlayAnimation(AnimState.Grab);
+        }
+        else if (!m_isMoving && m_stillTime > 0.25f && (m_toolSwitcher.HandFree))
+        {
+            PlayAnimation(AnimState.Idle);
+        }
+        // Petting animation priority should go here
+        else 
+        {
+            // No animation
+            PlayAnimation(AnimState.Stationary);
+        }
```

Comment:
after all the overengineering we've done this year, I'm more than happy with this approach lol - although I am curious why it isn't just playing the idle animation all the time it's not doing other activities. Like why does it need to be still for a period before the idle anim plays? I feel like it'd be fine to keep it going, but that might be a designer question

Reply:
The time delay is due to some weird behaviour with the PlayerHand's ismoving check. I think it's because it's checking every frame (which is like 50+ times per second), the mouse isn't actually moving on each and every frame, only some of them. What this means is that (and you can rest this with debug.log statements) is that when you move the mouse, it rapidly switches between moving and not moving. Without the delay, this would cause PlayerHandAnimationController to constantly Play idle/stationary/idle/stationary etc. which could look very jank.

Reply:
ahh gotcha no worries then